app/src/main/java/com/kickstarter/viewmodels/ResetPasswordViewModel.kt

```diff
@@ -0,0 +1,109 @@
+package com.kickstarter.viewmodels
+
+import com.kickstarter.libs.ActivityViewModel
+import com.kickstarter.libs.Environment
+import com.kickstarter.libs.rx.transformers.Transformers
+import com.kickstarter.libs.utils.StringUtils
+import com.kickstarter.models.User
+import com.kickstarter.services.ApiClientType
+import com.kickstarter.services.apiresponses.ErrorEnvelope
+import com.kickstarter.ui.activities.ResetPasswordActivity
+import rx.Observable
+import rx.subjects.PublishSubject
+
+interface ResetPasswordViewModel {
+
+  interface Inputs {
+    /** Call when the email field changes. */
+    fun email(emailInput: String)
+
+    /** Call when the reset password button is clicked. */
+    fun resetPasswordClick()
+  }
+
+  interface Outputs {
+    /** Emits a boolean that determines if the form is in the progress of being submitted. */
+    fun isFormSubmitting(): Observable<Boolean>
+
+    /** Emits a boolean that determines if the form validation is passing. */
+    fun isFormValid(): Observable<Boolean>
+
+    /** Emits when password reset is completed successfully. */
+    fun resetSuccess(): Observable<Void>
+
+    /** Emits when password reset fails. */
+    fun resetError(): Observable<String>
+  }
+
+  class ViewModel(val environment: Environment) : ActivityViewModel<ResetPasswordActivity>(environment), Inputs, Outputs {
+    private val client: ApiClientType = environment.apiClient()
+
+    private val email = PublishSubject.create<String>()
+    private val resetPasswordClick = PublishSubject.create<Void>()
+
+    private val isFormSubmitting = PublishSubject.create<Boolean>()
+    private val isFormValid = PublishSubject.create<Boolean>()
+    private val resetSuccess = PublishSubject.create<Void>()
+    private val resetError = PublishSubject.create<ErrorEnvelope>()
+
+    val inputs: Inputs = this
+    val outputs: Outputs = this
+
+    init {
+      this.email
+        .map { StringUtils.isEmail(it) }
```

Comment:
one cool FP trick you can do here is to make this function call point free, e.g.
```
        .map(StringUtils::isEmail)
```

Reply:
Thanks for pointing this out! I read about pointful vs point-free style.